modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -918,12 +918,14 @@ std::string serializeToBuffer(const LMDBBackend::LMDBResourceRecord& value)
   buffer.reserve(sizeof(len) + len + sizeof(value.ttl) + sizeof(value.auth) + sizeof(value.disabled) + sizeof(value.ordername));
 
   // Store the size of the resource record.
+  buffer.resize(sizeof(len));
   std::memcpy(&buffer.at(0), &len, sizeof(len));
```

Comment:
initially forgetting the resize here is in fact a fine example of why memcpy is the bad approach

Reply:
> this seems like a downgrade from .assign

Well, yes and no.

- This avoids the C-style cast, and reinterpret_cast which clang-tidy complains about.
- For this particular use-case, I believe cppreference even suggests std::memcpy.
- It is more explicit about how we're "serializing data" rather than "constructing a string".
- One downside is of course forgetting to resize beforehand as you mention in the other comment.

I'll resolve the other comment and we can have the discussion here.

Reply:
> When it is needed to interpret the bytes of an object as a value of a different type, [std::memcpy](https://en.cppreference.com/w/cpp/string/byte/memcpy) or [std::bit_cast](https://en.cppreference.com/w/cpp/numeric/bit_cast)(since C++20) can be used:

https://en.cppreference.com/w/cpp/language/reinterpret_cast

Reply:
> It is more explicit about how we're "serializing data" rather than "constructing a string".

To clarify this. `std::string` is absolutely the wrong type to use for this sort of thing. We should definitely move to `std::vector<uint8_t>` for this sort of thing. But that's a different topic and will probably never happen.

Reply:
> https://en.cppreference.com/w/cpp/language/reinterpret_cast

right, with a static_assert, on a separate line. We can't assert statically and it's easy to forget. memcpy makes me -very- uneasy